RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -122,7 +122,7 @@ public RestClient.Response getSnapshotStatus(String repoName, String snapshotNam
         String targetPath = "_snapshot/" + repoName + "/" + snapshotName;
         return client.getAsync(targetPath)
             .flatMap(resp -> {
-                if (resp.code == HttpURLConnection.HTTP_OK) {
+                if (resp.code == HttpURLConnection.HTTP_OK || resp.code == HttpURLConnection.HTTP_NOT_FOUND) {
```

Comment:
Shouldn't the 404 case be handled differently or at least logged?

Reply:
Confirmed - we throw at a higher level.  After the other changes in this PR, we'll also return an empty `Optional` in this case.